helm/ngrok-operator/templates/agent/deployment.yaml

```diff
@@ -117,9 +121,9 @@ spec:
         volumeMounts:
         {{ toYaml .Values.extraVolumeMounts | nindent 10 }}
         {{- end }}
-        {{- if .Values.lifecycle }}
+        {{- if or .Values.lifecycle $agent.lifecycle }}
         lifecycle:
-        {{ toYaml .Values.lifecycle | nindent 10 }}
+        {{ toYaml (default $agent.lifecycle .Values.lifecycle) | nindent 10 }}
```

Comment:
Let's undo these 2 changes since we don't have an agent specific lifecycle configuration yet.